app/frontend/src/utils/hooks.test.ts

```diff
@@ -26,3 +28,79 @@ describe("useSafeState hook", () => {
     expect(result.current[0]).toBe(2);
   });
 });
+
+describe("useGeocodeQuery hook", () => {
+  beforeAll(() => {
+    server.listen();
+  });
+  afterEach(() => {
+    server.resetHandlers();
+  });
+  afterAll(() => {
+    server.close();
+  });
+
+  it("works with expected loading state and result", async () => {
+    const { result, waitFor } = renderHook(() => useGeocodeQuery());
+    expect(result.current).toMatchObject({
+      isLoading: false,
+      error: undefined,
+      results: undefined,
+      query: expect.anything(),
+    });
+    await act(() => result.current.query("test"));
+    expect(result.all[1]).toMatchObject({
```

Comment:
`result.current` here is already the final result, any ideas? Or is this fine?

Reply:
Maybe don't await on the act and use `await waitForNextUpdate()`?

Otherwise a little obscure for reading, but it's probably ok as it at least shows the states the hook went through from start to finish in `result.all`

Reply:
Didn't work :(